hivemq-edge/src/main/java/com/hivemq/fsm/ProtocolAdapterFSM.java

```diff
@@ -116,53 +120,61 @@ public void stopAdapter() {
     }
 
     public boolean transitionAdapterState(final @NotNull AdapterStateEnum newState) {
-        final var currentState = adapterState.get();
-        if(canTransition(currentState, newState)) {
-            if(adapterState.compareAndSet(currentState, newState)) {
-                log.debug("Adapter state transition from {} to {} for adapter {}", currentState, newState, adapterId);
-                notifyListenersAboutStateTransition(currentState());
-                return true;
+        while (true) {
```

Comment:
This seems to be a busy wait. I suggest we add some sleep, otherwise, 1 CPU core will reach 100% for a while.